query/tests/it/common/hashtable.rs

```diff
@@ -63,3 +68,46 @@ fn test_two_level_hash_table() {
 
     assert_eq!(entity.get_value(), &2);
 }
+
+#[test]
+fn test_hash_map_drop() {
+    #[derive(Debug, Clone)]
+    struct A {
+        item: Arc<AtomicUsize>,
+    }
+
+    impl A {
+        pub fn new(item: Arc<AtomicUsize>) -> Self {
+            item.fetch_add(1, Ordering::Relaxed);
+            Self { item }
+        }
+    }
+
+    impl Drop for A {
+        fn drop(&mut self) {
+            self.item.fetch_sub(1, Ordering::Relaxed);
+        }
+    }
+
+    let item = Arc::new(AtomicUsize::new(0));
+
+    {
+        let mut table = HashMap::<u64, Vec<A>>::create();
+        for _ in 0..1200 {
+            // generate random key without external crate
+            let num1 = vec![2, 3];
+            let key = (&num1 as *const Vec<i32>) as u64 % 88;
```

Comment:
`num1` is a stacked value in loop and the alignment of `Vec<i32>` guarantees `key` is a multiple of 8. It's not random in my environment.

Reply:
Nice catch!